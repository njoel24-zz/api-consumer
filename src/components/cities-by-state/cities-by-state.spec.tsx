import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from "enzyme";

import Cities, { CitiesByStateProps } from './cities-by-state.component';

describe('City by state', () => {
	const cities = ['Milan', 'Turin', 'Naples'];
	const getWeatherPollutionByCity = jest.fn();

	it('renders component and compare with snapshot', () => {
		const component = renderer.create(<Cities cities={cities}/>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render links', () => {
		const wrapper = mount(
			<Cities cities={cities} />
		);
		const links = wrapper.find('a');
		expect(links.length).toBe(3);
	});

	it('should print cities', () => {
		const wrapper = mount(
			<Cities cities={cities}/>
		);
		wrapper.find('a').forEach((singleLink) => {
			expect(singleLink.text()).not.toBeNull();
		});
	});

	it('should ask for detail by clicking on a link', () => {
		const wrapper = mount(
			<Cities cities={cities} getWeatherPollutionByCity={getWeatherPollutionByCity}/>
		);
		const firstLink = wrapper.find("a:first-child");
		firstLink.simulate("click");
		expect(getWeatherPollutionByCity).toBeCalled();
	});
});