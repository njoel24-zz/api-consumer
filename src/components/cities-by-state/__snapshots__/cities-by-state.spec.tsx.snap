// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City by state renders component and compare with snapshot 1`] = `
<div>
  <a
    className=""
    onClick={[Function]}
  >
    Milan
  </a>
  <a
    className=""
    onClick={[Function]}
  >
    Turin
  </a>
  <a
    className=""
    onClick={[Function]}
  >
    Naples
  </a>
</div>
`;
